pages/LivePage.tsx

```diff
@@ -14,21 +14,38 @@ import { JSX } from "preact/jsx-runtime";
 import { PreactComponent } from "$live/engine/block.ts";
 import { CONTENT_SLOT_NAME } from "$live/sections/Slot.tsx";
 import UseSlot from "$live/sections/UseSlot.tsx";
+import { VNode } from "preact";
+import LivePagePreview, {
+  SectionControls,
+} from "$live/components/LivePagePreview.tsx";
 
 export interface Props {
   name: string;
   layout?: Page;
   sections: Section[];
 }
+
+export function renderSection(
+  props: Props["sections"][0],
+  idx: number,
+): JSX.Element;
+export function renderSection(
+  props: Props["sections"][0],
+  idx: number,
+  controls: VNode | false,
+): JSX.Element;
 export function renderSection(
   { Component: Section, props, metadata }: Props["sections"][0],
   idx: number,
+  controls: VNode | false = false,
 ) {
   return (
+    // data-manifest-key used at preview
     <section
       id={`${metadata?.component}-${idx}`}
       data-manifest-key={metadata?.component}
     >
+      {controls}
       <Section {...props} />
     </section>
   );
```

Comment:
since you're receiving all the properties you need for render the controls `<SectionControls>` in this function how about partial apply this function with the `preview` boolean and then create an empty element when preview is false? e.g

```ts
const renderSectionFor = (preview: boolean = false) => {
  const Controls = preview ? SectionControls : () => null;
  return (
    { Component: Section, props, metadata }: Props["sections"][0],
    idx: number,
  ) => {
    return (
      <section
        id={`${metadata?.component}-${idx}`}
        data-manifest-key={metadata?.component}
      >
        <Controls {...metadata} index={idx} />
        <Section {...props} />
      </section>
    );
  };
};
```

Then you can just use it directly `sections.map(renderSectionFor(preview))`

Reply:
and you can also export the default renderSection 

export const renderSection = renderSectionFor();

Reply:
Fixed